web/packages/teleport/src/Discover/Desktop/index.tsx

```diff
@@ -46,6 +47,7 @@ export const DesktopResource: Resource = {
   views: [
     {
       title: 'Select Resource',
+      eventName: DiscoverEvent.ResourceSelection,
```

Comment:
@ryanclark @marcoandredinis i think we need these additional event names (i copied and pasted the file and put a comment marker to see what i'm talking about, since it's hard to see with git diff).

i am unsure what to do with steps `Connect Teleport` and `Discover Desktops`. i think these two steps are combined under `Deploy Service` for other resources  (eg `DownloadScript` or `HelmChart`). could be that desktops works slightly different?

```ts
views: [
    {
      title: 'Select Resource',
      eventName: DiscoverEvent.ResourceSelection,
    },
    {
      title: 'Install Active Directory',
      component: InstallActiveDirectory,  // <------- need event here, user has to manually do stuff here (sim to like database mutualTLS)
    },
    {
      title: 'Connect Teleport',
      component: ConnectTeleport,
      eventName: DiscoverEvent.DeployService,     // <---- wasn't sure of this one, this step you create a join token, and user has to run it?
    },
    {
      title: 'Discover Desktops',
      component: DiscoverDesktops,   // <------- this step is where we "poll" for desktops
    },
    {
      title: 'Finished',
      component: Finished,
      eventName: DiscoverEvent.Completed,
      hide: true,
    },
  ],
```

Reply:
For Kube/HelmChart, I think we could use the DeployService step
Seems equivalent, right?

Sorry, I didn't knew how Discover Desktop works.
I agree that we need more events.
Your mappings look correct.

New event for Install Active Directory :+1: 

Connect Teleport seems to match the DeployService event.
Given that we have 4 sub-steps, the user can leave at any point, which could be useful to understand how easy the guide is.
To know where the user stopped or got an error, we could either create 4 distinct events or add some extra info to the error message in case of `error` or `abort` status.
What do you think?
Pinging @ryanclark as well.
If we end up with 4 steps, IMO, we should use the DeployService anyway as the latest step (so we would only create 3 new event names)

For the Discover Desktops I agree that it should be a new event.
Would it be useful to have some extra info? Like how many resources were discovered?
Maybe on a 2nd iteration

Reply:
Agreed on adding more steps to cover the missing pages

> If we end up with 4 steps, IMO, we should use the DeployService anyway as the latest step (so we would only create 3 new event names)

Agreed on this, so we can keep the last step consistent across the board

> For the Discover Desktops I agree that it should be a new event.
> Would it be useful to have some extra info? Like how many resources were discovered?
> Maybe on a 2nd iteration

It wouldn't hurt to add this info as we already have it for free.

Reply:
@marcoandredinis after talking with @ryanclark, we need the following events, we have substeps within steps for desktops:

```ts
views: [
    {
      title: 'Select Resource',
      eventName: DiscoverEvent.ResourceSelection,
    },
    {
      title: 'Install Active Directory',
      component: InstallActiveDirectory,  // <------- need event 
    },
    {
      title: 'Connect Teleport',
      component: ConnectTeleport,
        // This step has substeps worth emitting events:
        // 1) script that configures active directory        <----- need event
        // 2) asking user to copy and paste outputted teleport config     // skipping, as we think it's not useful to emit
        // 3) asking user to create `teleport.yaml` and copy and paste   // skipping
        // 4) deploy the desktop service                     <----- use existing `DiscoverEvent.DeployService` 
    },
    {
      title: 'Discover Desktops',
      component: DiscoverDesktops,          // <------- need event (waiting for desktops to be discovered)
    },
    {
      title: 'Finished',
      component: Finished,
      eventName: DiscoverEvent.Completed,
      hide: true,
    },
  ],
```

so we'll need (you can change the name)
- DiscoverEvent.InstallActiveDirectory
- DiscoverEvent.ConfigureActiveDirectory
- DiscoverEvent.DiscoverDesktops

Reply:
done wiring here: https://github.com/gravitational/teleport/pull/20581/commits/dd6f2647e83feb7d8fc9e999af6180fb7d849340

Reply:
What do you think about a generic Discovered Resources?
After looking into the new Database Flow, it seems we could re-use it there

Reply:
done: https://github.com/gravitational/teleport/pull/20581/commits/74a142dda9fdb071c4ea8d9f6345d8f828730e77